src/mcts/search.cc

```diff
@@ -366,6 +366,26 @@ std::pair<Move, Move> Search::GetBestMove() const {
   return GetBestMoveInternal();
 }
 
+bool Search::PopulateRootMoveLimit(MoveList* root_moves) const {
+  // Search moves overrides tablebase.
+  if (!limits_.searchmoves.empty()) {
+    *root_moves = limits_.searchmoves;
+    return false;
+  }
+  auto board = played_history_.Last().GetBoard();
+  if (!syzygy_tb_ || !board.castlings().no_legal_castle() ||
+      (board.ours() + board.theirs()).count() > syzygy_tb_->max_cardinality()) {
+    return false;
+  }
+  if (!syzygy_tb_->root_probe(played_history_.Last(), root_moves)) {
+    if (!syzygy_tb_->root_probe_wdl(played_history_.Last(), root_moves)) {
```

Comment:
Why nested ifs  `if (!a) if (!b) return false; ` and not just `if (!a && !b) return false` ?
Or even `return a || b;`

Reply:
The later option seems reasonable, I'll do that.